\documentclass[12pt]{article}
\linespread{1.5}
\addtolength{\oddsidemargin}{-2.5cm}
%\addtolength{\evensidemargin}{-3cm}
\addtolength{\textwidth}{2.5cm}
\begin{document}

\title{Making multiple-locus association mapping on a genome-wide scale routine}
\author{Andrew W. George and Joshua Bowden}

\maketitle


\abstract{
In this paper, we present our method that makes multiple locus association mapping routine. 
It is only a little more complicated than performing a linear mixed model analysis. 
Yet, we are able to fit all marker-trait associations simultaneously while still clearly identifying
 which marker-trait associations are true. There are no thresholds to be set, no multiple 
 testing issues,  and no parameters to fine tune. We will show that our method is capable 
 of  finding marker-trait associations more accurately,  and in many cases, orders of 
 magnitude faster than competing approaches. Our multiple locus method is implemented 
 in a freely available, fully documented and maintained R package called am+.
}

\section{Introduction}

Gwas for many are now the design of choice when interested in elucidating the genetic 
underpinnings  of complex traits.  Over the past decade, gwas have changed 
considerably in focus and scope. Early gwas 
were performed predominantly as case control studies. Their focus 
was on identifying the casual variants underlying common disorders in humans. 
Despite many promising findings, reproducibility of these findings 
proved an issue (refs). With the collection of dense whole genome data challenging, 
early gwas lacked power (refs).  Today, gwas are performed not just in humans but 
animals and plants that may have highly structured pedigrees. 
They can involve hundreds to thousands of individuals with genotypes collected 
from thousands to hundreds of thousands of snps. Where early gwas were interested
 in disease traits, modern gwas are also  interested in  categorical and quantitative traits.


Over the past decade, there have also been impressive advancements in genome wide 
association mapping methods, the statistical techniques for measuring the strength of 
association between a marker locus and trait. Traditionally, association data were 
analysed with simple contingency tests or Cochran-armitage  trend tests. Today, far 
more statistically sophisticated methods are available. One such method is the  QK 
approach, a linear mixed model framework for genome wide association 
mapping (refs).  Several highly efficient computer programs for performing QK-based 
association mapping are available. Programs such as EMMA, fastLMM, GEMMA, and 
XXX(ref) make feasible the analysis of data from hundreds of thousands of snps, collected 
on thousands of subjects. However, not all areas of genome-wide association 
mapping have advanced. It still remains standard practice to analyse genome-wide 
association data with a single locus model. That is,  it is common for the strength of 
association between a marker locus and trait is measured for each locus separately. 

Methods for association mapping with multiple locus models have been developed 
for some time. Computer implementations of these methods have also been available. 
Thus, it is not a lack of methodology  or computational tools preventing the uptake of 
multiple locus association mapping. The reasons for their lack of acceptance are many. 
Multiple locus association mapping methods are often cased in high level statistical theory, 
making their true value difficult to appreciate.  Computationally, they can scale poorly with 
sample size and/or number of genetic markers. Some methods require tuning parameters 
to be specified. These methods also report their results as coefficients whose significance 
are unknown. There are multiple locus methods that instead report their results as probabilities 
but the difficulty here is determining  an appropriate threshold for inferring a statistically 
significant finding. The computer implementations can also be difficult to run in some cases. 
All these reasons contribute to making multiple locus association mapping unappealing as 
an analysis method.

In this paper, we present our new am+ approach for multiple locus association mapping. 
Our approach is based on linear mixed models and uses the same parameter estimation 
procedure as EMMA. Through a clever dimension reduction step, our approach scales 
with number of subjects instead of number of marker loci. We build the multiple locus 
model iteratively. At each step, only a single linear mixed model is fitted to the data. 
This model fits all marker loci and their association with the trait simultaneously. There 
are no parameters to be tuned, no thresholds to be set. Our approach automatically 
finds the best model. Once our approach is run, the set of statistically significant 
marker loci in strongest association with the trait is returned as its findings.  

Our am+ approach is implemented in a fully documented and maintained R package. 
We are able to achieve impressive performance by moving the computationally 
components of our algorithm to Cpp. Here, we make full use of distributed 
computing, through multiple cpu and multiple glue processing when available. 
Our R package is easy to use, even for non  R users. It consists of a single function
 to read in the data and a single function for analysis. We will demonstrate that 
 our approach returns more accurate results faster and more efficiently than 
 competing multiple locus association mapping approaches. Our am+ approach 
 makes multiple locus association mapping of genome wide data routine.

\section{Results}

\subsection{Computer implementations for Multiple Locus Association Mapping}



We have chosen six computer implementations whose performance we  
compared  to our R package, am+. These are bigRR, glmnet, , lmm-lasso,
MLMM,  piMASS, and randfor.  We chose 
these six implementations because a. they were purpose-built for 
multiple-locus association mapping or contained general methods from 
which multiple-locus association mapping techniques had been developed, 
b. they could handle analysis on a genome wide scale, c.  they were freely 
available, and d. they are state-of-the-art. 

A summary of the key 
features of the programs/packages whose 
performance we evaluated in this paper are given in table X.
Even from this small set of 
packages and computer programs, the diversity of implementation is apparent. 
BigRR and glmnet are based 
on linear models, lmm-lasso and MLMM on linear mixed models, piMASS on Bayesian models, 
and randfor on decision tree models. BigRR fit its models to data with ridge regression, a regularisation technique. 
Glmnet and lmm-lasso instead rely on lasso where parameter estimation and variable selection occur simultaneously. 
PiMass fits its models with bayesian variable selection, MLMM through stepwise selection, and randfor with random forests. 
Some of the implementations evaluated here fit all snp simultaneously where others do not. Although the underlying statistical 
problem was the same, to find the "best" set of predictions (snps) to explain a quantitative response (trait), the implementations differed greatly. 

\subsection{Simulation study to evaluate computational performance}

\subsection{Simulation study to evaluate statistical performance}

\subsection{Application to Arabidopsis data}


\end{document}
