\documentclass{article}


\begin{document}

\title{Making multiple-locus association mapping on a genome-wide scale routine}
\author{Andrew W. George and Joshua Bowden}

\maketitle


\abstract{
In this paper, we present our method that makes multiple locus association mapping routine. It is only a little more complicated than 
performing a linear mixed model analysis. Yet, we are able to fit all marker-trait associations simultaneously while still clearly identifying
 which marker-trait associations are true. There are no thresholds to be set, no multiple testing issues,  
 and no parameters to fine tune. We will show that our method is capable of  finding marker-trait associations more accurately, 
 and in many cases, orders of magnitude faster than competing approaches. Our multiple locus method is implemented 
 in a freely available, fully documented and maintained R package called am+.
}

\section{Introduction}

Gwas for many are now the design of choice when interested in elucidating the genetic underpinnings  
of complex traits.  Over the past decade, gwas have changed considerably in focus and scope. Early gwas 
were performed predominantly as case control studies. Their focus was on identifying the casual variants 
underlying common disorders in humans. Despite many promising findings, reproducibility of these findings 
proved an issue (refs). With the collection of dense whole genome data challenging, early gwas lacked power (refs).  
Today, gwas are performed not just in humans but animals and plants that may have highly structured pedigrees. 
They can involve hundreds to thousands of individuals with genotypes collected from thousands to hundreds of
 thousands of snps. Where early gwas were interested in disease traits, modern gwas are also  interested in 
 categorical and quantitative traits.


Over the past decade, there have also been impressive advancements in genome wide association mapping methods, the statistical techniques for measuring the strength of association between a marker locus and trait. Traditionally, association data were analysed with simple contingency tests or Cochran-armitage  trend tests. Today, far more statistically sophisticated methods are available. One such method is the  QK approach, a linear mixed model framework for genome wide association mapping (refs).  Several highly efficient 
computer programs for performing QK-based association mapping are available. Programs such as EMMA, fastLMM, GEMMA, and 
XXX(ref) make feasible the analysis of data from hundreds of thousands of snps, collected on thousands of subjects. However, not all 
areas of genome-wide association mapping have advanced. It still remains standard practice to analyse genome-wide association data 
by measuring the strength of association between a marker locus and trait for each locus separately. 




Over the past decade, there have also been impressive advancements in genome wide association mapping methods. Association mapping methods are statistical techniques for measuring the strength of association between a marker locus and trait. It has long been recognised that spurious marker-trait associations can occur due to population structure. This problem has now been solved (refs). A linear mixed model framework has also been shown to be well suited to analysing data from gwas. Linear mixed models can accommodate multiple sources of variation simultaneously, including population structure and familial relatedness (refs). Fitting lmm's to gwas data though was computationally challenging. This limitation has also been solved via spectral decomposition (ref) and approximation (ref) leading to several very fast lmm implementations (refs). Despite these impressive advancements, a core feature of mainstream  association mapping has remained unchanged. The strength of association between a maker locus and trait is assessed via a single locus model.  That is, the trait and marker data from gwas are analysed for each marker locus separately.

Interestingly, it isn't a lack of computational tools that is restricting the adoption of multiple locus association mapping.  Several  computer programs are available. Glmnet, lmmlasso, and lars are R packages that implement shrinkage methods for regression problems.   All marker-trait associations are fitted simultaneously with the coefficients of loci with spurious association to a trait shrunk to zero. Another R package that was purpose built for the multiple locus analysis of data from gwas is the bigRR package.  It implements generalised ridge regression. Similar to shrinkage methods, all marker trait associations are fitted simultaneously. However, coefficients are not shrunk and  determining their significance  can be challenging. PiMASS is a command line driven program that implements Bayesian variable selection for multiple locus association mapping. The strength of the marker-locus association is measured by a posterior probability. Parameter inferences are obtained via Markov Chain Monte Carlo.  MLMM is an R package .. linear mixed model that is implemented extremely well. 

\end{document}
