\documentclass[12pt]{article}

\linespread{1.5}
\addtolength{\oddsidemargin}{-2.5cm}
%\addtolength{\evensidemargin}{-3cm}
\addtolength{\textwidth}{2.5cm}
\begin{document}

\title{Making multiple-locus association mapping on a genome-wide scale routine}
\author{Andrew W. George and Joshua Bowden}

\maketitle


\abstract{
The methods of choice for genome wide association mapping all share a common shortcoming. 
They measure the strength of marker-trait associations on a locus by locus basis. 
More sophisticated multi-locus methods that map multiple marker-trait associations jointly are available. 
However, these methods are computationally demanding, are not always easy to implement, and can give results that are difficult to interpret. The aim of this paper is to present our method that makes the multi-locus analysis of data from genome wide association studies routine. It is formulated within a linear mixed model framework. We call our method AMplus. 
AMplus produces results faster than competing multi locus methods and often with greater statistical power. 
In terms of speed, AMplus is just a little slower than the fastest single-locus linear mixed model implementations. 
AMplus is freely available as a fully documented R package which we port as a docker binary for platform independent portability.
}


\section{Introduction}

Over the past decade, g-w association mapping methods have changed considerably. Early mapping methods relied upon 
simple contingency tables or logistic regression for case-control designs and score tests for family-based designs. 



Over the past decade, genome-wide association mapping methods, have changed considerably. Early association mapping 
methods were no more complicated than contingency table tests or logistic regression. This is because early association 
studies followed case-control designs. A significant difference in the allele frequency at a locus, between the cases (affecteds) and 
controls (unaffecteds), was interpreted as the locus being in linkage disequilibrium with a casual gene. 
It became known though that differences in allele frequency could also arise due to unrecognised population stratification. 
This prompted a shift towards family-based designs. Family-based designs do not suffer from 
the confounding affects of population stratification. A myriad of new family-based association mapping methods were developed, 
the most popular based on score tests (ref for TDT S-TDT Spielman and Ewens 1998, AJHG 62:450-458, FBAT Naird, NM; Horvath, S; Xu, X 2000, genetic epi V19, Suppl 1 S36-S42). 













Over the past decade, genome-wide association mapping, the statistical techniques for analysing data from association studies, has advanced considerably. Early genome-wide association studies (GWAS) began as case-control designs. The premise was that casual genes could be detected via differences in allele frequency between cases (affecteds) and controls (unaffecteds). Association mapping methods for these designs were no more sophisticated than contingency table tests or logistic regression. Unfortunately, differences 
in allele frequency can also arise due to unrecognised population stratification. Consequently, there was a shift towards family-based 
designs. Findings from these designs are
not susceptible to the confounding affects of population stratification. With interest growing in genome-wide association studies, 
a myriad of different association mapping methods for family-based designs were developed. 


Over the past decade, there have also been impressive advancements in genome wide 
association mapping methods, the statistical techniques for measuring the strength of 
association between a marker locus and trait. (Needs more detail - what are some of the early assoc study methods (give a single sentence review).... TDT, genomic control, structured association, score tests???? 
Traditionally, association data were 
analysed with simple contingency tests or Cochran-armitage  trend tests. Today, far 
more statistically sophisticated methods are available. One such method is the  QK 
approach, a linear mixed model framework for genome wide association 
mapping (refs).  The QK approach 


Several highly efficient computer programs for performing QK-based 
association mapping are available. Programs such as EMMA, fastLMM, GEMMA, and 
XXX(ref) make feasible the analysis of data from large GWAS. 
 However, not all areas of genome-wide association 
mapping have advanced. It still remains standard practice to analyse genome-wide 
association data with a single locus model. That is,  it is common for the strength of 
association between a marker locus and trait is measured for each locus separately. 

Methods for association mapping with multiple locus models have been developed 
for some time. Computer implementations of these methods have also been available. 
Thus, it is not a lack of methodology  or computational tools preventing the uptake of 
multiple locus association mapping. The reasons for their lack of acceptance are many. 
Multiple locus association mapping methods are often cased in high level statistical theory, 
making their true value difficult to appreciate.  Computationally, they can scale poorly with 
sample size and/or number of genetic markers. Some methods require tuning parameters 
to be specified. These methods also report their results as coefficients whose significance 
are unknown. There are multiple locus methods that instead report their results as probabilities 
but the difficulty here is determining  an appropriate threshold for inferring a statistically 
significant finding. The computer implementations can also be difficult to run in some cases. 
All these reasons contribute to making multiple locus association mapping unappealing as 
an analysis method.

In this paper, we present our new am+ approach for multiple locus association mapping. 
Our approach is based on linear mixed models and uses the same parameter estimation 
procedure as EMMA. Through a clever dimension reduction step, our approach scales 
with number of subjects instead of number of marker loci. We build the multiple locus 
model iteratively. At each step, only a single linear mixed model is fitted to the data. 
This model fits all marker loci and their association with the trait simultaneously. There 
are no parameters to be tuned, no thresholds to be set. Our approach automatically 
finds the best model. Once our approach is run, the set of statistically significant 
marker loci in strongest association with the trait is returned as its findings.  

Our am+ approach is implemented in a fully documented and maintained R package. 
We are able to achieve impressive performance by moving the computationally 
components of our algorithm to Cpp. Here, we make full use of distributed 
computing, through multiple cpu and multiple glue processing when available. 
Our R package is easy to use, even for non  R users. It consists of a single function
 to read in the data and a single function for analysis. We will demonstrate that 
 our approach returns more accurate results faster and more efficiently than 
 competing multiple locus association mapping approaches. Our am+ approach 
 makes multiple locus association mapping of genome wide data routine.

\section{Results}


\subsection{Computer programs for   multiple locus association mapping}

We compared the performance of our R package, am, to six other computer programs for multiple locus association mapping. These programs are ..., ..., ... We decided on these six because they implement published methods for multiple locus association mapping.  are capable of analysing association data on a genome wide scale, and are freely available. A summary of each of these programs along with am+ is given in table X. In preparing this table, we assumed the purpose of analysing data from a genome wide association study is to identify those snp in true association with a trait.  From table X, the computer programs vary greatly in methodology, implementation, requirements, and utility.  For example, some programs have methodology that fits all marker-trait associations simultaneously where others  fit only a subset of marker-trait associations jointly. Some programs have tuning parameters where others have none. Several cannot handle additional fixed effects such as covariates to account for population structure. Also, the majority of programs considered here yield results that require additional computation before a "best" set of snp is realised. Only am+  fits all marker trait associations simultaneously, handles different input data, accommodates additional fixed effects, has no parameters to be fine tuned, returns the "best" set of snp in joint association with a trait without additional computation, and is easy to use. 



\subsection{Computational performance}
We conducted a simulation study to assess  the computational performance of am+. For comparison, we assessed six other computer programs (table X) that  also implement multiple locus association mapping. The marker data in our simulation study comes from phase 3 of the 1000 genomes project. Here, 2315 individuals were sequenced and their genotype information made available. We generated trait data by randomly selecting snp, assigning allelic  effects to the selected snp, and aggregating these effects along with random error. The number of snp  selected  was sampled from a Poisson distribution with mean 30. The effect sizes were sampled from a beta distribution. The errors were drawn from a normal distribution with zero mean and whose variance yielded a quantitative trait with a heritability of 0.5.


We were interested in how study size impacts performance. We generated replicates under five different study sizes. These were 150 individuals and 4000 snp (150 X 4K), 300 individuals and 50000 snp (300 X 50k), ... We based these five study sizes on the size of study being performed in unsequenced crop plants such as wheat, ......, respectively. For those study sizes that exceeded the number of genotyped individuals in the 1000 genomes project,  we simulated additional data by forming genotype mosaics from the real snp data. We generated 250 replicates for each scenario. 

Performance was measured by recording elapse  time and maximum memory usage. Analyses were performed on a high end desk top which was a dell intra quad core processor with 120G of RAM and a Intel gpu. Programs were run for a maximum of 100 hours.  All programs except piMass  made use of distributed computing, either implicitly via multithreaded blas and lapac linear algebra libraries or as in the case of ranfor, explicitly. For ranfor, we wrote R code to parallelise the random forest analysis across multiple cpu.  All scripts for performing these analyses are available from the Csiro-data acces portal.

In figure X we compare the elapse time of  am+ against the other computer programs. On the y axis, we have the ratio of each computer program's elapse time to the elapse time of cpu-based am+. On the X axis, we have the number of genotypes and study size. Both axes are on log scale . A positive (negative) log ratio means am+ had a shorter (longer) elapse time  than the other computer program. The magnitude (value) of the ratio is (the magnitude of)  how many times shorter or longer am+ is to the other computer program.  Since performance varies with replicate, for a given study size, a point in figure X is the log median of 250  ratios of elapse times.


From figure X, the superior performance of am+ to the other computer programs is evident. Describe ...


\subsection{Simulation study to evaluate computational performance}

\subsection{Simulation study to evaluate statistical performance}

\subsection{Application to Arabidopsis data}


\end{document}
